### PR TITLE
docs(core): Update README.md

### PR DESCRIPTION
**Pull request by sephmard**
Description
-----------

- Fixes WB-19488

What does the PR do? Include a concise description of the PR contents.

Updated readme for wandb-core. Aligning with /wandb/wandb repo.

- [ ] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

File: core/README.md [hunk 1]
```diff
@@ -1,22 +1,149 @@
...
+Get started with wandb-core in four steps:
+
+1. First, sign up for a [free W&B account](https://wandb.ai/login?utm_source=github&utm_medium=code&utm_campaign=wandb&utm_content=quickstart).
+
+2. Second, install the W&B SDK with [pip](https://pip.pypa.io/en/stable/). Navigate to your terminal and type the following command:
+
+```bash
+pip install wandb
+```
+***Note: ensure you have `wandb>=0.17.0`.***
```
Comment (dmitryduev) [inline core/README.md:105]: Let's update this to the latest version and add this file to the bumpversion config.

Comment (sephmard) [reply core/README.md]: Updated. But not sure how to include a bumpversion config to maintain it automatically

Comment (dmitryduev) [reply core/README.md]: add an entry for the readme file here: https://github.com/wandb/wandb/blob/main/.bumpversion.wandb.cfg

Comment (sephmard) [reply core/README.md]: Ok: https://github.com/wandb/wandb/pull/7919

File: core/README.md [hunk 2]
```diff
@@ -1,22 +1,149 @@
...
+
+# Save model inputs and hyperparameters in a wandb.config object
+config = run.config
+config.learning_rate = 0.01
+
+# Model training code here ...
+
+# Log metrics over time to visualize performance with wandb.log
+for i in range(10):
+    run.log({"loss": loss})
```
Comment (kptkin) [inline core/README.md:138]:
```suggestion
    run.log({"loss": loss})
    
   run.finish(
```

**Review by kptkin — APPROVED** [review]
Nice! Great improvement

File: core/README.md [hunk 3]
```diff
@@ -1,22 +1,149 @@
...
 ```
 
-Note: ensure you have `wandb>=0.17.0`.
 
-### Platform compatibility
+
+
+## Compatibility
```
Comment (kptkin) [inline core/README.md:144]: we wanted to make the table collapsible

Comment (sephmard) [reply core/README.md:144]: I looked in to making this collapsable and its easy, but the issue is how GIthub renders the markdown footnotes. The footnotes are rendered outside of the collapsable menu. I think the footnotes help with readability, so opted not to make this collapsable.

Comment (kptkin) [reply core/README.md:146]: Ah good to know. Yeah in that case I think it’s better to keep as is

Comment (sephmard) [reply core/README.md:146]: Actually, I think I figured it out. Updated!

File: core/README.md [hunk 4]
```diff
@@ -1,22 +1,151 @@
...
+config = run.config
+config.learning_rate = 0.01
+
+# Model training code here ...
+
+# Log metrics over time to visualize performance with wandb.log
+for i in range(10):
+    run.log({"loss": loss})
+    
+run.finish(
```
Comment (dmitryduev) [inline core/README.md:140]:
```suggestion
run.finish()
```

**Review by dmitryduev — APPROVED** [review]
